### PR TITLE
Fix PTM branch traversal to prevent finding too many involved atoms

### PR DESCRIPTION
**Pull request by pckroon**
Fixes part of #142
Sem-Ver: bugfix

### CODE REVIEW

File: vermouth/processors/canonicalize_modifications.py [hunk 1]
```diff
@@ -78,35 +78,34 @@ def find_ptm_atoms(molecule):
...
-            if orig in extra_atoms:
+        while True:
+            succ = molecule[orig].keys()
+            if orig in extra_atoms and orig not in atoms:
                 # If this is a PTM atom, we want to see it's neighbours as
                 # well.
                 to_see.update(succ)
                 atoms.add(orig)
-            else:
+            elif orig not in extra_atoms and orig not in anchors:
```
Comment (Tsjerk) [inline vermouth/processors/canonicalize_modifications.py]: 'and orig not in anchors' makes some sense for following the logic, but omitting the check does not change the result of the code (adding an element to a set that already contains that element); I think I'd prefer removing it as it then follows that any 'orig' that failed the first test is an anchor and added to the set as such.

File: vermouth/processors/canonicalize_modifications.py [hunk 2]
```diff
@@ -78,35 +78,34 @@ def find_ptm_atoms(molecule):
...
-        to_see = set([first])
+        to_see = set()
         # Traverse in molecule.
-        for orig, succ in nx.bfs_successors(molecule, first):
-            # We've seen orig, so remove it
-            if orig in to_see:
-                to_see.remove(orig)
-            if orig in extra_atoms:
+        while True:
+            succ = molecule[orig].keys()
```
Comment (Tsjerk) [inline vermouth/processors/canonicalize_modifications.py]: You only use 'succ' within the first if-block. Why not put it there?

**Comment by pckroon** [comment]
Travis currently fails on building docs, which is fixed in #206. So merge that first.

**Comment by pckroon** [comment]
Or, I just cherry-pick the appropriate commit.

**Pull request merged**